src/services/PrestoService.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+namespace lewiscom\presto\services;
+
+use Craft;
+use craft\helpers\DateTimeHelper;
+use lewiscom\presto\Presto;
+use craft\base\Component;
+use craft\helpers\FileHelper;
+use lewiscom\presto\records\PrestoCachePurgeRecord;
+use yii\db\Query;
+
+class PrestoService extends Component
+{
+    /**
+     * @var mixed
+     */
+    public $settings;
+
+    /**
+     * @var
+     */
+    public $rootPath;
+
+    /**
+     * PrestoService constructor.
+     *
+     * @param array $config
+     */
+    public function __construct(array $config = [])
+    {
+        parent::__construct($config);
+
+        $this->settings = Presto::getInstance()->getSettings();
+        $this->rootPath = $this->settings->rootPath;
+    }
+
+    /**
+     * Write the HTML output to a static cache file
+     *
+     * @param string $host
+     * @param string $path
+     * @param string $html
+     * @param array $config
+     * @throws \yii\base\ErrorException
+     */
+    public function writeCache($host, $path, $html, $config = [])
+    {
+        if (! Craft::$app->user->isGuest) {
+            return;
+        }
+
+        // TODO: Check if this check is necessary
```

Comment:
Did this check check get checked? Or is this to check in the future on checking the check check?